### PR TITLE
✨ ocm transport

### PR DESCRIPTION
**Pull request by nirrozenbaum**
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by nirrozenbaum** [comment]
I modified transport interface function `WrapObjects` to return `runtime.Object`.
I don't want every transport implementation to have to convert to unstructured, but rather I have implemented the conversion to unstructured inside the generic implementation.
this way, ocm specific returns manifest work as is, and every other transport would return its wrapped object with no need to convert.

**Review by vMaroon — APPROVED** [review]
LGTM, can merge after handling the failed check

File: pkg/transport/provider/ocm/go.mod [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	github.com/modern-go/reflect2 v1.0.2 // indirect
+	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/text v0.7.0 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448 // indirect
+	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
+)
```
Comment (shmuelk) [inline pkg/transport/provider/ocm/go.mod:26]: This file shouldn't be here. It should be under ocm-transport or transport-ocm or transport/ocm

File: pkg/transport/provider/ocm/go.sum [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+k8s.io/klog/v2 v2.80.1/go.mod h1:y1WjHnz7Dj687irZUWR/WLkLc5N1YHtjLdmgWjndZn0=
+k8s.io/utils v0.0.0-20221128185143-99ec85e7a448 h1:KTgPnR10d5zhztWptI952TNtt/4u5h3IzDXkdIMuo2Y=
+k8s.io/utils v0.0.0-20221128185143-99ec85e7a448/go.mod h1:OLgZIPagt7ERELqWJFomSt595RzquPNLL48iOWgYOg0=
+open-cluster-management.io/api v0.12.0 h1:sNkj4k2XyWA/GLsTiFg82bLIZ7JDZKkLLLyZjJUlJMs=
+open-cluster-management.io/api v0.12.0/go.mod h1:/CZhelEH+30/pX7vXGSZOzLMX0zvjthYOkT/5ZTzVTQ=
+sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 h1:iXTIw73aPyC+oRdyqqvVJuloN1p0AC/kzH07hu3NE+k=
+sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2/go.mod h1:B8JuhiUyNFVKdsE8h686QcCxMaH6HrOAZj4vswFpcB0=
+sigs.k8s.io/structured-merge-diff/v4 v4.2.3 h1:PRbqxJClWWYMNV1dhaG4NsibJbArud9kFxnAMREiWFE=
+sigs.k8s.io/structured-merge-diff/v4 v4.2.3/go.mod h1:qjx8mGObPmV2aSZepjQjbmb2ihdVs8cGKBraizNC69E=
+sigs.k8s.io/yaml v1.3.0 h1:a2VclLzOGrwOHDiV8EfBGhvjHvP46CtW5j6POvhYGGo=
```
Comment (shmuelk) [inline pkg/transport/provider/ocm/go.sum:81]: This file shouldn't be here. It should be under ocm-transport or transport-ocm or transport/ocm

File: pkg/transport/provider/ocm/ocm.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		Spec: workv1.ManifestWorkSpec{
+			Workload: workv1.ManifestsTemplate{
+				Manifests: manifests,
+			},
+		},
+	}
+
+	return manifestWork
+
+}
```
Comment (shmuelk) [inline pkg/transport/provider/ocm/ocm.go:60]: This file shouldn't be here. It should be under ocm-transport or transport-ocm or transport/ocm and there under pkg....

**Pull request merged**